simd/src/Kokkos_SIMD_Common.hpp

```diff
@@ -92,17 +96,34 @@ class where_expression<bool, T> : public const_where_expression<bool, T> {
 };
 
 template <class T, class Abi>
-[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
-    where_expression<simd_mask<T, Abi>, simd<T, Abi>>
-    where(typename simd<T, Abi>::mask_type const& mask, simd<T, Abi>& value) {
+[[nodiscard]] KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION std::enable_if_t<
+    !std::is_same_v<Abi, Kokkos::Experimental::simd_abi::scalar>,
+    where_expression<simd_mask<T, Abi>, simd<T, Abi>>>
+where(typename simd<T, Abi>::mask_type const& mask, simd<T, Abi>& value) {
+  return where_expression(mask, value);
+}
+
+template <class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<
+    std::is_same_v<Abi, Kokkos::Experimental::simd_abi::scalar>,
```

Comment:
Why SFINAE rather than a non-templated `simd_abi::scalar` overload?

Reply:
Mostly, because it feels right. These functions are doing the same thing with identical implementation. Also, it looks clearer than replacing `Abi` everywhere.

I checked anyway that we can do a function overload instead.